src/pages/RepoPage/BundlesTab/BundleContent/BundleSummary/hooks/useBundleSummary.ts

```diff
@@ -0,0 +1,61 @@
+import { useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import { useBranches } from 'services/branches'
+import { useRepoOverview } from 'services/repo'
+
+import { useBundleBranchSelector } from './useBundleBranchSelector'
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export const useBundleSummary = () => {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const [branchSearchTerm, setBranchSearchTerm] = useState<string>()
+
+  const { data: overview, isLoading } = useRepoOverview({
+    provider,
+    owner,
+    repo,
+  })
+
+  const {
+    data: branchList,
+    isFetching: branchListIsFetching,
+    hasNextPage: branchListHasNextPage,
+    fetchNextPage: branchListFetchNextPage,
+  } = useBranches({
+    repo,
+    owner,
+    provider,
+    filters: { searchValue: branchSearchTerm },
+    opts: {
+      suspense: false,
+    },
+  })
+
+  const { data: branchesData, fetchNextPage: branchesFetchNextPage } =
+    useBranches({ repo, owner, provider })
```

Comment:
Confused as to why we are calling this hook twice? Can we search just once with a default filter value?

Reply:
I believe this is the PR that added it to the Coverage tab `useSummary` hook: https://github.com/codecov/gazebo/pull/1859

Reply:
Err maybe this one: https://github.com/codecov/gazebo/pull/1803